_posts/2022-03-07-fine-dining-for-indices.md

```diff
@@ -0,0 +1,408 @@
+
+## Please Ingest Responsibly!
+
+OpenSearch will index every single field encountered in an ingested document. If a new field is ingested, OpenSearch will do its best to determine the data type and update the index mapping accordingly. This is called “Dynamic Mapping” and is also the default behavior. This is where being intimate with your data set is very helpful. Being too granular can cause what is known as a **mapping explosion**, where the number of fields in an index causes storage and retrieval to be bound and slow. 
+
+If your diet is particularly strict, you can ignore fields that are not part of an explicit mapping. Consider the previous example with another option added:
+
+
+```
+PUT explicit_mapping_index
+{
+    "mappings": {
+    "dynamic" : false,
+        "properties": {
+            "eat_me": {
+                "type": "text"
+            },
+            "drink_me": {
+                "type": "text"                
+            },
+            "dont_index_me": {
+                "type": "text",
+                "index": false
+            }
+        }
+    }
+}
+```
+
+Note the addition of `dynamic: false`. With this option in place, your mapping will never be automatically updated when ingesting a field that has not been seen before.  Let's repeat our previous experiment with this new **explicitly mapped** index. 
+
+```
+POST explicit_mapping_index/_doc
+{
+	"eat_me": "iron ration",
+	"drink_me": "potion of index familiarity",
+	"dont_index_me": "I'm expecting this field to be ingested, but because I'm familiar with my data I know beforehand I don't need it. It will not be indexed and won't be returned in search queries. It will be retrievable in the `_source` field if this document is ever retrieved.",
+        "some_other_field": "This field is not in the mapping to exemplify Dynamic Mapping."
+}
+```
+
+Remember with the **dynamically mapped** index, a new entry was added to the mapping when a new field was ingested. Checking our mapping now shows that `some_other_field` was not added automatically. 
+
+```
+GET explicit_mapping_index/_mapping
+{
+  "explicit_mapping_index" : {
+    "mappings" : {
+      "dynamic" : "false",
+      "properties" : {
+        "dont_index_me" : {
+          "type" : "text",
+          "index" : false
+        },
+        "drink_me" : {
+          "type" : "text"
+        },
+        "eat_me" : {
+          "type" : "text"
+        }
+      }
+    }
+  }
+}
+```
+
+## Can I Keep the Recipe? 
+
+By default the entire document ingested is saved as a field named `_source`. This field can be disabled (it is by default enabled) when creating your index,  usually for storage space efficiency.  Ingested documents are indexed, but the originally submitted document is not stored. In this case, if you decide not to index every field encountered, the data will not be part of the index, nor will it be retrievable from the `_source` field. 
```

Comment:
On second read, I'm a little iffy calling `_source` a field. It's an object.

So: 
'ingested is saved as a field named `_source`. This field can be disabled' -> 'ingested to the `_source` object. `_source` can be disabled'

and

'nor will it be retrievable from the `_source` field. ' -> 'nor will it be retrievable from the `_source` object.'